pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -766,3 +767,54 @@ func setProviderConfig(rawConfig RawConfig, s v1alpha1.ProviderConfig) (*runtime
 
 	return &runtime.RawExtension{Raw: rawPconfig}, nil
 }
+
+func (p *provider) SetInstanceNumberForMachines(machines v1alpha1.MachineList, metrics *prometheus.GaugeVec) error {
+	if len(machines.Items) < 1 {
+		return nil
+	}
+	config, _, _, err := p.getConfig(machines.Items[0].Spec.ProviderConfig)
```

Comment:
We assume that the credentials are identical for all machines. This means that rotating the credential will break this metric until the machine at position zero in the slice has the appropriate one.

The alternative to that would be to maintain a `map[credentialsHash]ec2Client` inside this func. Do you think that should be done?

Reply:
I have implemented that as a `map[identifier]credentials`